gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/AmbiguousStopStationTransfersValidatorTest.java

```diff
@@ -0,0 +1,223 @@
+          .setCsvRowNumber(5)
+          .setFromStopId("stop3")
+          .setToStopId("stationB")
+          .build();
+  private final GtfsTransfer transfer_from_boarding_area1_in_stationA_to_stationB =
+      new GtfsTransfer.Builder()
+          .setCsvRowNumber(6)
+          .setFromStopId("boarding_area1")
+          .setToStopId("stationB")
+          .build();
+  private final GtfsTransfer transfer_from_stationA_to_entrance2_in_stationB =
+      new GtfsTransfer.Builder()
+          .setCsvRowNumber(7)
+          .setFromStopId("stationA")
+          .setToStopId("entrance2")
+          .build();
+
+  @Test
+  public void sameStationSameDirectionTransfersShouldGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    List<GtfsStop> stops =
+        new ArrayList<>(Arrays.asList(stationA, stationB, stop1_in_stationA, stop2_in_stationB));
+    validator.stopTable = GtfsStopTableContainer.forEntities(stops, noticeContainer);
+    List<GtfsTransfer> transfers =
+        new ArrayList<>(
+            Arrays.asList(
+                transfer_from_stop1_in_stationA_to_stationB,
+                transfer_from_stationA_to_stop2_in_stationB));
+    validator.transferTable = GtfsTransferTableContainer.forEntities(transfers, noticeContainer);
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices())
+        .containsExactly(
+            new AmbiguousStopStationTransfersNotice(
+                /* transferCsvRowNumberA= */ 1,
+                /* transferCsvRowNumberB= */ 2,
+                /* fromStationId= */ "stationA",
+                /* toStationId= */ "stationB"));
+  }
+
+  @Test
+  public void sameStationDifferentDirectionTransfersShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    List<GtfsStop> stops =
+        new ArrayList<>(Arrays.asList(stationA, stationB, stop1_in_stationA, stop2_in_stationB));
+    validator.stopTable = GtfsStopTableContainer.forEntities(stops, noticeContainer);
+    List<GtfsTransfer> transfers =
+        new ArrayList<>(
+            Arrays.asList(
+                transfer_from_stop1_in_stationA_to_stationB,
+                transfer_from_stop2_in_stationB_to_stationA));
+    validator.transferTable = GtfsTransferTableContainer.forEntities(transfers, noticeContainer);
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+
+  @Test
+  public void differentStationTransfersShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    List<GtfsStop> stops =
+        new ArrayList<>(
+            Arrays.asList(
+                stationA,
+                stationB,
+                stationC,
+                stop1_in_stationA,
+                stop2_in_stationB,
+                stop3_in_stationC));
+    validator.stopTable = GtfsStopTableContainer.forEntities(stops, noticeContainer);
```

Comment:
Why not just add all of the stops in a shared class constructor (as well as having a common noticeContainer)?

Reply:
This stops' combination is only used once in our tests - are we still preferred to create a shared class constructor for that?

Reply:
If I read the tests correctly, each test adds some subest of stops to the stops table, and having more stops in the stops table would not have any impact. The logic of the tests is about the transfer table.
I am suggesting removing the common fixture from the test.

Reply:
Good advice! I have made the shared stops and also a common noticeContainer in the latest version.